src/nitypes/bintime/_timedelta_array.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+
+from collections.abc import Collection, Iterable, MutableSequence
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    final,
+    overload,
+)
+
+import numpy as np
+import numpy.typing as npt
+
+if TYPE_CHECKING:
+    # Import from the public package so the docs don't reference private submodules.
+    from nitypes.bintime import CVITimeIntervalDType, TimeDelta, TimeValueTuple
+else:
+    from nitypes.bintime._dtypes import CVITimeIntervalDType
+    from nitypes.bintime._timedelta import TimeDelta
+    from nitypes.bintime._time_value_tuple import TimeValueTuple
+
+
+@final
+class TimeDeltaArray(MutableSequence[TimeDelta]):
+    """An array of :class:`TimeDelta` values in NI Binary Time Format (NI-BTF).
+
+    The TimeDeltaArray class provides a mutable sequence container for storing
+    multiple :class:`TimeDelta` objects efficiently.
+
+    This class implements the MutableSequence interface, providing familiar
+    list-like operations for time delta collections.
+
+    :param value: Initial collection of TimeDelta objects, defaults to empty array
+    :type value: Collection[TimeDelta] | None, optional
+
+    :raises TypeError: If any item in value is not a TimeDelta instance
+
+    Examples:
+        Create an empty array:
+
+        >>> arr = TimeDeltaArray()
+        >>> len(arr)
+        0
+
+        Create with initial values:
+
+        >>> from nitypes.bintime import TimeDelta
+        >>> td1 = TimeDelta(seconds=1.5)
+        >>> td2 = TimeDelta(seconds=2.5)
+        >>> arr = TimeDeltaArray([td1, td2])
+        >>> len(arr)
+        2
+
+        Access and modify elements:
+
+        >>> arr[0]
+        nitypes.bintime.TimeDelta(Decimal('1.5'))
+        >>> arr[0] = TimeDelta(seconds=3.0)
+        >>> arr[0]
+        nitypes.bintime.TimeDelta(Decimal('3'))
+
+    See Also:
+        * :class:`TimeDelta`: Individual time delta values
+    """
+
+    __slots__ = ["_array"]
+
+    _array: npt.NDArray[np.void]
+
+    def __init__(
+        self,
+        value: Collection[TimeDelta] | None = None,
+    ) -> None:
+        """Initialize a new TimeDeltaArray.
+
+        :param value: Initial collection of TimeDelta objects
+        :type value: Collection[TimeDelta] | None, optional
```

Comment:
Parameter documentation should be English prose. The documentation already includes types from the type hints.